test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -240,6 +240,19 @@ public void ParseUnquotedValue ()
 
             Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("0\n1"));
             Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("'"));
+            Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("\""));
+            Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse(" "));
+            Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("\t"));
+
+            // if you don't go to End(), these do not attempt to consume newlines or comments
+            Assert.Equal("0", Parsers.UnquotedValue.Parse("0\n1").Value);
```

Comment:
As per discussion on the alternative PR https://github.com/tonerdo/dotnet-env/pull/82#discussion_r1437274187 this already worked before any of these changes, and is simply a consequence of removing the `End()` which allows the parser to consume as much as it can, rather than attempting to consume the entire string.

Reply:
In general I would say that the Tests without "End" are nearer to the actual cases at the end of the day, because in "reality" we have no "End" except for the last value, and this is only true if the file ends with the value and not  with a comment, a newline or a comment-only line. And even with that there is no End-Call on the value itself, just on the `Assignment.[...].AtLeastOnce()`.

But both types of tests definitely help to make the system more robust when introducing code-changes of course, so nothing to complain about the End-Tests, just the reson why I like having tests which assure that it also works if there follows additional input.

Reply:
An example to demonstrate it:

With this new test (with End) we check, that singlequotes are not allowed inside unquotedValues:
`Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("a'b'c"));`

We are able to add another test:
`Assert.Equal("a", Parsers.UnquotedValue.Parse("a'b'c").Value);`
Same content to be parsed, and now you see, that "in reality" this Parser will not throw on this input, but instead it will return `a` and hands over to the next parser.
That is not a problem currently in this solution, because the next Parsers within Assignment will fail because they do not expect an input of `'b'c`, but it leads to exceptions-messages from a different parser.

In this case this is okay I guess, but this is due to the fact, that we do not define a "valid ending" for unquoted values (compared to quotedValues, but that's way more easy there of course...)